### DISCUSSION TITLE
Basics of the basics

### DISCUSSION THREAD

**Discussion by Plasma-Rob** [shannon-lab/zapdos#177] [Q&A Getting Started]
I'm interested in modeling plasma from a physics and chemistry perspective.  Is Zapdos an appropriate option?  My first question is can I get an estimated ion density between two plasma sources if I know the gas, pressure, power, and the geometry?  If so, where do I start to learn how to accomplish this?  I'd like to do 1D, then 2D.  Any thoughts on where to get started would be helpful.

**Comment by csdechant** [comment]
Yes, I believe Zapdos is a appropriate option for modeling plasma physics and chemistry. Zapdos (along with coupling other physics modules within the MOOSE ecosystem, like [CRANE](https://github.com/lcpp-org/crane)) has be used for simulating plasma-water interaction for atmospherics plasma in 1-D and CCP plasma at mid-low pressures (0.1 - 1 Torr) in 1-D & 2-D. Here is a link to previous [publication](https://shannon-lab.github.io/zapdos/publications.html), with the exception of a few papers currently under review. Some of the current research with Zapdos involves microwave discharges and atmospheric plasma jets.

For the question about ion density; yes, Zapdos can estimate the ion density. The only caveat being that Zapdos currently does have power has in input parameter, but instead, it uses an input voltage. That being said, Zapdos is open-source and is continually being developed on, so any missing physics or capabilities can be added by individual user or addressed in the Issues tab of GitHub.

For where to begin with Zapdos, we have a tutorial directory within Zapdos. These tutorial have user change different parameters within the input files to see how that affects different physical phenomena. For additional information, there are several files in the doc directory mentioned in [README.md](https://github.com/shannon-lab/zapdos#readme). 

@cticenhour please let me know if I may have miss any other source of information for Zapdos, or if you would like to add anything.

**Reply by cticenhour** [reply]
You pointed out the Zapdos website for publications, but I want to put the main link here as well: https://shannon-lab.github.io/zapdos. Getting started documentation for installation and object-level documentation can be found here. We've still got a lot of work to do in terms of making a guided written tutorial in the website, but the tutorial files @csdechant mentioned are a good place to start. Please feel free to reach out with questions as you come across them - we prefer you make a new discussion thread for any unique question you have. We'd like this forum to eventually become a good resource for new users alongside our tutorial and other documentation. 

Thank you for your interest!

**Reply by Plasma-Rob** [reply]
Thank you both for taking time to respond.  I have everything installed.  I can and have run the tutorials, but I have not been able to customize them to my use case of interest.  The one that might be a potential to use is tutorial 3 that does the ion-only plasma, but the point in that example deals with the exiting mass and that is where my interest deviates.  

The install docs are great.  Beyond that I have found it very challenging to decipher how to set up a problem that isn't premade in a tutorial.

tutorial 1 is diffusion.  it refers to lieberman's figure 5.1 pg 141.  (where is this? Is it "Principles of Plasma Discharges and Materials Processing?")
tutorial 2 is a simple reaction and again refers to a text figure 9.1 on pg 291 again with no reference.
tutorial 3 is ion only plasma and looks like it might be the closest to my interest at the moment
tutorial 4 is the timing test to see how good your system is... 400s is the fastest I've heard of on a standalone machine
tutorial 5 is the plasma water interface.  

I'm not interested in water right now.  All of the papers referenced in the documentation are dealing with liquid.  Is zapdos the right package or should I stick with crane?

Having voltage as an input can work since I have power, voltage, and current values for system I would like to model and it would be fair to state that being able to ramp the voltage up and down may have an effect.  How does one set up a spatial region and declare one side powered (voltage) and look at the ion density as a function of position in 1D?  To me this means generating a mesh to represent the spatial region, but that is where the questions begin.

Thanks again for taking a minute to help.

**Reply by csdechant** [reply]
Yes, the Lieberman's figures refers to the figures in "Principles of Plasma Discharges and Materials Processing".

Tutorial 4 isn't meant to test how good your system is. It is 1D argon CCP discharge, which the user is suppose to change the pressure and pressure dependent parameters (such as  mobility and diffusivity) to see how the plasma changes based on pressure. Just curious, what from the input file or documents made you think that tutorial was for hardware testing (as @cticenhour stated, we have lot of work to do in terms of making a guided written tutorial, so I just want to make sure where we can clarify things)?

CRANE only deals with chemistry networks. It is great for 0-D chemistry problems, but CRANE doesn't hold any of the plasma fluid physics (Zapdos contains the plasma fluid physics and electrostatic solver for the potential).  Mesh generation for Zapdos can be found on the Zapdos website [here](https://shannon-lab.github.io/zapdos/getting_started/zapdos_meshing.html) (I personal use [Gmsh](https://gmsh.info/) to construct meshes for Zapdos).

What kind of discharge is your system and what gases are you using? Depending on your setup, you might be able to modify tutorial 4. As is, it is a solves for the ions, electrons, electron energy, and potential for a 1-D argon CCP discharge at 1 Torr. For boundary conditions, it uses the thermal flux for the electron and electron energy, the advection flux for the ions, and the potential is driven at 13.56 MHz 100V peak to peak at one end and grounded at the other end. To view the output files, we recommend using [Paraview](https://www.paraview.org/).

**Reply by Plasma-Rob** [reply]
Thank you for the clarification and the links and again, thank you for taking time to respond.   This is very helpful.  I've looked at the zapdos mesh and I'll order the book.  I know tutorial 4 isn't the timing test... thanks for humoring me.  S.S's M2 did it in 410s though, so I've used it as a speed test to see if my system is commensurate.  I'll scour that example more thoroughly.

Made a mesh with gmsh...  probably overkill for a 1d case, but not for a 2D case.  How do I plug that 1D mesh into a capacitive argon plasma at 10 mTorr with 13.56mHz and say 1000 V with ground on one end?  What are the other available options included in the program for types of discharge besides capacitive?

I'll run it again and use paraview to see the changes.  I'm running zapdos on a standalone via ssh, so I have to push the model file and pull the data to view.  Is there a recommended way to do that?

**Reply by csdechant** [reply]
To answer your question a little out of order:

- To change the pressure to 10mTorr in Tutorial 4, you would need to change all pressure dependent values. This includes the background gas density (lines 139-146), the electron material properties (line 228 - 248), and ion mobility and diffusivity (lines 255 - 256). All of these relationships are assumed to be linear with pressure and currently they are set to 1Torr (so all the values need to change by a factor of 100).
- To change to voltage in Tutorial 4, you need to change the function of the voltage on line 150 -153. The voltage for this simulation has a function Dirichlet BC on the applied voltage side (which function currently represents a driven voltage at 13.56 MHz 100V peak to peak in the form of $V = 0.100kV * \sin(2\pi * 13.56e6Hz * t)$ . The ground voltage on the other side is done by applying a Dirichlet BC set to zero.
- If you want to look at changing the mesh, I would recommend modifying Tutorial 6. The reason being is that Tutorial 4 reads its mesh information and initial conditions from a previous output file 'RF_Plasma_WithOut_Metastables_IC.e'. Tutorial 6 is the same as Tutorial 4 (a CCP argon plasma) but has less preset inputs, so user has more options to change stuff (like the meshing). The mesh information is located in the Mesh Block (lines 19 - 39 in 'RF_Plasma_Blank.i' in Tutorial 6).
- In the past, we have mainly looked at CCP (such as DC pin-to-plate or RF parallel-plate like the GEC reference source) for verification and validation. We also have capability for 1-D DBD discharges and surface charge effects (we don't have a tutorial for that, but the test file is /test/tests/surface_charge/dbd_mesh.i). By coupling in other MOOSE modules (such as the electromagnetic and Navier-Stokes module), we are currently looking at microwave discharges and plasma jets. To clarify, there is no direct option within Zapdos to declare a discharge type. Discharge types are define by what type of boundary conditions are used. For more information on the difference boundary types within Zapdos, I would recommend looking at the [Zapdos PDF file](https://github.com/shannon-lab/zapdos/tree/devel/doc/tutorial_pdf). Also, since Zapdos is based on the MOOSE framework and if you are not familiar with the layout of MOOSE, I would recommend chapter 3 of Alex Lindsay's [PhD dissertation](https://github.com/shannon-lab/zapdos/blob/devel/doc/Lindsay_thesis_coupling_of_plasmas_and_liquids.pdf) and the [MOOSE tutorial videos](https://mooseframework.inl.gov/getting_started/examples_and_tutorials/index.html).
- If you are using a ssh, does the server you are connecting to house any visualization options to read exodus file and can you use any GUI options (like xrdp)? If not, you can try coping the output life to your local machine with the `scp` commend.

**Reply by Plasma-Rob** [reply]
This is great.  Thank you! A couple easy follow up questions and then I'll need to do some testing.

1.  Pressure is changing by a factor of 10, not 100, is it a square function?  Asking for when I change it to something else.
2.  What does BC stand for?

Thanks for the links to the PhD dissertation and videos.  
The plan was to use scp to transfer the output files to the local machine for visualization.  Paraview downloaded...

**Reply by csdechant** [reply]
For your questions:

1. The default pressure of Tutorial 4 is 1 Torr.  In the previous set of questions, it was asked to change it to 10 mTorr. That should be a factor of 100 decrease in pressure (unless there was a typo). 
2. BC stands for Boundary Condition.

Hope testing goes well. As @cticenhour stated before, please start a new discussion thread for any unique question you have.

**Reply by Plasma-Rob** [reply]
Both super obvious questions.  Thanks for stooping to answer.  I think in mTorr, so that is where 10 vs 100 comes from...  

I think the major answer to all my questions for fundamentals is the moose tutorial.  It is 11 hours long and I'm only a couple hours in, but it shows where zapdos is and how this process is set up to work.  I did not understand how zapdos calls on moose and this tutorial helps show that with another package.  I did not understand that relationship clearly, which led to my zapdos specific question, which was really a moose question.  The tutorial illustrates the process to set up the input file and the importance of fundamentally understanding the governing equations.  Thank you for answering my questions.
